crm_timesheet/tests/test_crm_phonecall.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# License AGPL-3 - See LICENSE file on root folder for details
+##############################################################################
+
+from . import base
+import datetime
+
+
+class CrmPhonecallCase(base.BaseCase):
+
+    def test_button_end_call(self):
+        date_10 = datetime.datetime.now() - datetime.timedelta(minutes=10)
+        cases = (
+            (False, 'name01', '2015-08-25 12:30:00', 0.0),
+            ('2015-08-21 12:30:00', 'name02', '2015-08-21 12:45:00', 15.0),
+            (date_10.strftime("%Y-%m-%d %H:%M:%S"), 'name03', None, 10.0),
+            ('2015-07-01 11:00:00', 'name04', '2015-07-01 10:55:00', -5.0)
+        )
+        for date, name, end_date, expected_duration in cases:
+            self.next_case()
+            phonecall = self.phonecall_create({'date': date, 'name': name})
+            phonecall.button_end_call(end_date=end_date)
+            self.assertAlmostEqual(phonecall.duration, expected_duration,
+                                   places=0)
+
+    def test_phonecall_create_write(self):
+        create_cases = (
+            {'date': '2015-08-21 10:15:00',
+             'duration': 15,
+             'name': 'test_01',
+             'analytic_account': self.analytic_account.id,
+             'partner_id': self.partner.id,
+             'user_id': self.uid},
+            {'date': '2015-08-21 10:15:00',
+             'duration': 30,
+             'name': 'test_02',
+             'analytic_account': self.analytic_account.id,
+             'partner_id': self.partner.id,
+             'user_id': self.uid}
+        )
+        write_cases = (
+            {'date': '2015-08-20 11:20:00'},
+            {'duration': 20},
+            {'partner_id': self.partner2.id},
+            {'analytic_account': self.analytic_account2.id},
+            {'duration': 0},
+            {'duration': -20},
+            {'partner_id': False},
+            {'analytic_account': False},
+            {'date': False}
+        )
+        for create_vals in create_cases:
+            self.next_case()
+            phonecall = self.phonecall_create(create_vals)
+            if phonecall.timesheets:
+                self.assertPhonecallTimesheet(phonecall,
```

Comment:
Very smart way to do this, but don't name the method with upper chars, because this is only a way to differentiate unittest2 methods from Odoo ones, and you're are not expanding an assert, but creating a shortcut to evaluate several asserts at once. Call it for example `check_phonecall_timesheet_asserts`